settings/R/write.settings.R

```diff
@@ -1,35 +1,18 @@
 ##' Takes in a settings object, performs a series of checks, fixes & updates settings and produces pecan.CHECKED.xml
 ##'
-##' @title Fix Deprecated Settings
+##' @title Write settings
 ##' @param settings settings list
-##' @return updated settings list
+##' @param outputfile the file name to write to
+##' @param outputdir the directory to write to
 ##' @author Ryan Kelly
 ##' @author Betsy Cowdery
 ##' @export write.settings
 
 
-write.settings <- function(settings, outputfile = "pecan.CHECKED.xml"){
-  library(XML)
-  library(PEcAn.DB)
-  library(PEcAn.utils)
-  
-  settings <- papply(settings, fix.deprecated.settings)
-  settings <- papply(settings, addSecrets)
-  settings <- papply(settings, update.settings)
-  settings <- check.settings(settings)
-  
-  ## save the checked/fixed pecan.xml
-  if (!is.null(outputfile)) {
-    pecanfile <- file.path(settings$outdir, outputfile)
-    if (file.exists(pecanfile)) {
-      logger.warn(paste("File already exists [", pecanfile, "] file will be overwritten"))
-    }
-    saveXML(listToXml(settings, "pecan"), file=pecanfile)
+write.settings <- function(settings, outputfile, outputdir=settings$outdir){
+  pecanfile <- file.path(outputdir, outputfile)
+  if (file.exists(pecanfile)) {
+    logger.warn(paste("File already exists [", pecanfile, "] file will be overwritten"))
   }
```

Comment:
per discussion below on workflow.R, would prefer it if you restored @bcow's previous function, rename it to something more intuitive, and then recreated the write.settings that you have now.